### ISSUE TITLE
"Sticky" sample is not sticky

### ISSUE THREAD

**Issue by the-not-mad-psychologist** [elmish/Elmish.WPF#456]
**High-level description**
The Sticky sample is supposed to update only when a new even number is given, but it updates with every increment, even or odd.

**Steps to Reproduce**

1. Run the Sticky sample.
2. Click on the "+" button three times.

**Expected or desired behavior**
The counter now reads "2", the most recent even number.

**Actual behavior**
The counter now reads "3", which is not even.

**Additional Information**
Elmish.WPF version: 4.0.0-beta-43
Target framework version: .NET 5.0

From `BindingInternal.BindingData.addSticky`:

```F#
let addSticky (predicate: 'model -> bool) (binding: BindingData<'model, 'msg>) =
  let mutable stickyModel = None
  let f newModel =
    match predicate newModel, stickyModel with
    | _, None ->
        newModel
    | true, _ ->
        stickyModel <- Some newModel
        newModel
    | false, Some sm ->
        sm
  binding |> mapModel f
```

`stickyModel` seems always to be set to `None`, so the `match` expression always returns `newModel` from the first branch. I hypothesize that `stickyModel` resets on every update.

How can we resolve this? Should we add another case to `BindingData`?

**Comment by TysonMN** [comment]
This used to work, so one approach to gain more information is to find the commit where this stopped working.

**Comment by TysonMN** [comment]
This is fixed in version [`4.0.0-beta-44`](https://www.nuget.org/packages/Elmish.WPF/4.0.0-beta-44).
